src/room_member.cpp

```diff
@@ -258,13 +262,8 @@ bool RoomMember::IsConnected() const {
 
 void RoomMember::Leave() {
     ASSERT_MSG(receive_thread != nullptr, "Must be in a room to leave it.");
-
-    {
-        std::lock_guard<std::mutex> lock(network_mutex);
-        peer->Shutdown(300);
-        state = State::Idle;
-    }
-
+    state = State::Idle;
```

Comment:
imo we should make sure that raknet is disconnected before setting the state to idle.
If we don't do that we might risk the case that our code thinks it's disconnected, when a connection is still alive in reality (which should not be acceptable / tolerable)